### PR TITLE
feat: add readme for AOD and template user guide

### PR DESCRIPTION
**Pull request by sqin2019**

### CODE REVIEW

File: README.md
Comment (yolocs) [inline README.md]:
```suggestion
This template is for creating an AOD (access on demand) repository to get access to GCP resources on demand.
```

File: README.md
Comment (yolocs) [inline README.md]:
```suggestion
## Make an AOD Request

**Make sure you have completed [prerequisite steps](link to the section below) to set up this repo properly.**

1. Open a PR
  - A title and description to provide enough justification. E.g. "AOD request: debug customer issue X".
  - To request IAM permissions on org/folder/project level, add an `iam.yaml` file in the repo root.
    - See an example of this file, see example in [example-iam.yaml](link to file).
    - Optionally use predefined duration labels on the PR to specify the IAM permission expiration. Otherwise a 2h default duration will be used.
  - (Later) To request on-demand `gcloud` commands, add an `gcloud.yaml` file in the repo root.

2. Checks
  - All AOD PRs will fail the check "do_not_merge" to prevent accidental merge.
  - The "Validate Request" check will report errors if there are problems in your `iam.yaml` or `gcloud.yaml`.
  
3. Getting approval
  - Ask one of the repo code owners to approve the PR
  - Upon approval, another GitHub workflow will run to grant you the IAM permissions or run the on-demand `gcloud` commands

3. Close the PR after you no longer need the access. The PR will automatically be closed after X hours (TODO).
```

File: README.md
Comment (yolocs) [inline README.md]:
```suggestion
## Prerequisites
```

Comment (yolocs) [inline README.md]: Add a section above about "How AOD works" and leave a TODO for me to fill in the content

File: README.md
Comment (yolocs) [inline README.md]: Add a section below about adding code other than AOD. Could be a follow up to fill in the content

File: README.md
Comment (yolocs) [inline README.md]: Can we reference some of existing module instead of dumping all the Terraform here?

File: README.md
Comment (yolocs) [inline README.md]: Define "who" should be carrying out these steps - likely someone with admin role on the project/folder/org

File: README.md
Comment (yolocs) [reply README.md]: Terraform also isn't easy to understand. How about we delete these for now and follow up. I'm thinking referencing https://github.com/google-github-actions/auth#setting-up-workload-identity-federation first then reference terraform.

File: README.md
Comment (yolocs) [inline README.md:45]: Also add a TODO to revamp the prerequisites

Comment (yolocs) [reply README.md:45]: Create some corresponding issues

**Review by yolocs — DISMISSED** [review]
Please add a bit PR description

**Review by yolocs — APPROVED** [review]

File: README.md
Comment (capri-xiyue) [inline README.md:52]: Set up  [Workload Identity Federation](https://cloud.google.com/iam/docs/workload-identity-federation) by following guide [here](https://github.com/google-github-actions/auth#setting-up-workload-identity-federation)?

Comment (capri-xiyue) [inline README.md:52]: Set up [Workload Identity Federation](https://cloud.google.com/iam/docs/workload-identity-federation) by following guide [here](https://github.com/google-github-actions/auth#setting-up-workload-identity-federation)?

File: README.md
Comment (capri-xiyue) [inline README.md:57]: The minimum permissions are `getIamPolicy` and `setIamPolicy` and these permissions can be on projects, folders, or organizations level.

File: README.md
Comment (capri-xiyue) [inline README.md:74]: Make `duration-4h` more obvious by changing the format like `duration-4h`

File: README.md
Comment (capri-xiyue) [inline README.md:82]: Add Code Owners link?

File: README.md
Comment (capri-xiyue) [inline README.md:88]: add CODEOWNERS link? and put it before `Require review from Code Owners`

File: README.md
Comment (capri-xiyue) [inline README.md:21]: The link showed 404.

File: README.md
Comment (capri-xiyue) [inline README.md:22]: (Optional)

File: README.md
Comment (capri-xiyue) [inline README.md:24]: Do we have an example of `gcloud.yaml`?

Comment (sqin2019) [reply README.md:24]: Not really, it is to be implemented. The design doc has a rough example, but since this feature is not available yet, plan to add that when it id done.

File: README.md
Comment (sqin2019) [reply README.md:52]: For now yes, we don't have a better guide for WIF yet.

Comment (capri-xiyue) [reply README.md:52]: I mean you can change the sentence to `Set up [Workload Identity Federation](https://cloud.google.com/iam/docs/workload-identity-federation) by following guide [here](https://github.com/google-github-actions/auth#setting-up-workload-identity-federation)`

**Pull request merged**
